Tests/SonarScanner.MSBuild.PreProcessor.Test/CacheProcessorTests.cs

```diff
@@ -249,6 +249,24 @@ public void ProcessPullRequest_SomeFilesChanged_ProducesOnlyUnchangedFiles()
             logger.AssertInfoLogged("Incremental PR analysis: 1 files out of 3 are unchanged.");
         }
 
+        [TestMethod]
+        public void ProcessPullRequest_UnexpectedCacheKeys()
+        {
+            using var sut = CreateSut(Mock.Of<IBuildSettings>(x => x.SonarScannerWorkingDirectory == @"C:\ValidBasePath"));
+            var cache = new AnalysisCacheMsg();
+            cache.Map.Add(new(Path.GetInvalidFileNameChars()), ByteString.Empty);
+            cache.Map.Add(new(Path.GetInvalidPathChars()), ByteString.Empty);
+            cache.Map.Add(string.Empty, ByteString.Empty);
+            cache.Map.Add("  ", ByteString.Empty);
+            cache.Map.Add("\t", ByteString.Empty);
+            cache.Map.Add("\n", ByteString.Empty);
+            cache.Map.Add("\r", ByteString.Empty);
```

Comment:
We can also use the initializer, with less code (but more lines). The focus will be on the actual values,
```suggestion
            var cache = new AnalysisCacheMsg
                        {
                            Map =
                            {
                                { new(Path.GetInvalidFileNameChars()), ByteString.Empty },
                                { new(Path.GetInvalidPathChars()), ByteString.Empty },
                                { string.Empty, ByteString.Empty },
                                { "  ", ByteString.Empty },
                                { "\t", ByteString.Empty },
                                { "\n", ByteString.Empty },
                                { "\r", ByteString.Empty },
                            }
                        };
```